src/core/services/theming/theming.spec.js

```diff
@@ -787,6 +798,38 @@ describe('md-theme directive', function() {
   });
 });
 
+describe('md-deferred-theme directive', function() {
+  beforeEach(module('material.core'));
+
+  it('should observe and set mdTheme controller', inject(function($compile, $rootScope) {
+    $rootScope.themey = 'unregistered';
+    var el = $compile('<div md-deferred-theme="{{themey}}">')($rootScope);
+    $rootScope.$apply();
+    var ctrl = el.data('$mdThemeController');
+    expect(ctrl.$mdTheme).toBe('unregistered');
+    $rootScope.$apply('themey = "blue"');
+    expect(ctrl.$mdTheme).toBe('blue');
+  }));
+
+  it('does not warns when an unregistered theme is used', function() {
+    inject(function($log, $compile, $rootScope) {
+      spyOn($log, 'warn');
+      $compile('<div md-deferred-theme="unregistered"></div>')($rootScope);
```

Comment:
You actually would want this to warn because the theme isn't registered. This approach loses that information.